internal/pkg/client/patch/patch_test.go

```diff
@@ -0,0 +1,164 @@
+`
+		mergePatch = `{"spec":{"template":{"spec":{"containers":[{"image":"nginx23","name":"nginx","resources":{}}]}}}}`
+		unstructuredDeployment = &unstructured.Unstructured{}
+		modifiedDeployment = &unstructured.Unstructured{}
+		scheme = kscheme.Scheme
+		scheme.Convert(dep, unstructuredDeployment, nil)
+		dep.Spec.Template.Spec.Containers[0].Image = "nginx23"
+		scheme.Convert(dep, modifiedDeployment, nil)
+		modifiedDeployment.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "apps",
+			Kind:    "Deployment",
+			Version: "v1",
+		})
+		unstructuredDeployment.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "apps",
+			Kind:    "Deployment",
+			Version: "v1",
+		})
+		close(done)
+	})
+
+	AfterEach(func(done Done) {
+		// Cleanup
+		close(done)
+	})
+
+	Describe("Patch", func() {
+		Context("GetModifiedConfiguration", func() {
+			It("should get correct serialization of incoming object", func(done Done) {
+				By("getting the modified config for deployment")
+				modifiedconfig, err := patch.GetModifiedConfiguration(unstructuredDeployment, true)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking expected modified config")
+				Expect(string(modifiedconfig)).To(Equal(annotatedSerializedDeployment))
+				close(done)
+			})
+
+			It("should get correct non-annotated serialization of incoming object", func(done Done) {
+				By("getting the modified config for deployment")
+				modifiedconfig, err := patch.GetModifiedConfiguration(unstructuredDeployment, false)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking expected modified config")
+				Expect(string(modifiedconfig)).To(Equal(serializedDeployment))
+				close(done)
+			})
+
+		})
+
+		Context("SetLastApplied", func() {
+			It("should annotate the lastapplied of incoming object", func(done Done) {
+				By("set last applied")
+				err := patch.SetLastApplied(unstructuredDeployment)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking expected annotation")
+				lastapplied, err := patch.GetLastApplied(unstructuredDeployment)
+				Expect(err).NotTo(HaveOccurred())
+				Expect((string(lastapplied))).To(Equal(serializedDeployment))
+				close(done)
+			})
+
+		})
+
+		Context("GetMergePatch", func() {
+			It("should return the correct patch forincoming object", func(done Done) {
+				By("get merge patch")
+				p, err := patch.GetMergePatch(unstructuredDeployment, modifiedDeployment)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(p.Type).To(Equal(types.MergePatchType))
+
+				By("checking expected patch")
+				Expect((string(p.Data))).To(Equal(mergePatch))
+				close(done)
+			})
+
+		})
+
+		Context("GetClientSideApplyPatch", func() {
```

Comment:
For `GetClientSideApplyPatch`, can you add the following tests:
- add a test for CRD to cover the unregistered type
- add a test without last applied annotation

Reply:
done.